(TESTING) Add autonomy logic

Isopod00 opened the pull request:
Add autonomy logic to the main_control_node. The main challenge of this task will be determining where to dig.

Isopod00 committed:
Initial Commit

Isopod00 committed:
Added 2 reminders

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

umnrobotics committed:
Added autonomus_field_type to rovr_control.yaml and coordinates to main_control_node

umnrobotics committed:
Added cycle condition for end_autonomous()

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Brady-Higgins committed:
started optimal_dig_location func

Isopod00 committed:
Merge branch 'main' into autonomy-logic

umnrobotics committed:
Changed optimal_dig_location function and location of the function

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

BergerKing333 committed:
theoretical way algorithm to find dig strips in test file

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
TF broadcaster improvements

Isopod00 committed:
Merge branch 'main' into reset-odom-improvements

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into reset-odom-improvements

Isopod00 committed:
Merge branch 'main' into autonomy-logic

BergerKing333 committed:
Swapped the field path to a config file parameter.

Isopod00 committed:
Finish the addition of the ROS param

BergerKing333 committed:
more work in test file

Isopod00 committed:
Merge remote-tracking branch 'origin/main' into autonomy-logic

Isopod00 committed:
Merge remote-tracking branch 'origin/main' into reset-odom-improvements

Isopod00 committed:
Merge branch 'main' into reset-odom-improvements

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Added  a frame_id_renamer so the apriltag TFs  can be visualized in rviz

umnrobotics committed:
Commented tings out and addded file save output

Isopod00 committed:
Merge branch 'main' into reset-odom-improvements

Isopod00 committed:
fixed robot starting rotated

Isopod00 committed:
Applied an offset so that odom starts at the zed

Isopod00 committed:
Merge branch 'main' into reset-odom-improvements

Isopod00 committed:
Added the basicnavigator

Isopod00 committed:
better starting spot on the field

Isopod00 committed:
reverted an old "fix" and pushing the part that works so far

Isopod00 committed:
current working state

Isopod00 committed:
minor change

Isopod00 committed:
working calibration routine for a parallel zed

Isopod00 committed:
improvements

Isopod00 committed:
delete testing code

Isopod00 committed:
minor change

Isopod00 committed:
remove unused imports

Isopod00 committed:
bring back translation_end _vector for easier accurate positioning

Isopod00 committed:
getting close

Isopod00 committed:
use the known map coordinates as an offset

Isopod00 committed:
no more averaging

Isopod00 committed:
remove unused import

Isopod00 committed:
actually use the orientation of the tags properly

Isopod00 committed:
Merge branch 'main' into reset-odom-improvements

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'reset-odom-improvements' into autonomy-logic

BergerKing333 committed:
digging pseudo code

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
some cleanup

Isopod00 committed:
Merge branch 'main' into autonomy-logic

BergerKing333 committed:
adding things

BergerKing333 committed:
moving to rov_control

BergerKing333 committed:
moving to rovr control + swapping to lineCost

Isopod00 committed:
Merge branch 'main' into autonomy-logic

BergerKing333 committed:
moving to while loop

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
increased this time because our robot is slow

Isopod00 committed:
deleted the weird combine_esdf file

Isopod00 committed:
minor changes

Isopod00 committed:
more correct

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
small fix

Isopod00 committed:
things work now

Isopod00 committed:
happy with this nav2 performance

Isopod00 committed:
auto format

Isopod00 committed:
minor improvements

Isopod00 committed:
more improvements

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

BergerKing333 committed:
creating a custom costmap2d object and using that to generate lineCost/ dig locations

jmblixt3 commented on src/rovr_control/rovr_control/main_control_node.py:
```diff
@@ -149,21 +151,30 @@ def __init__(self) -> None:
+        resolution = costmap.getResolution()
+
+        # NEEDED MEASUREMENTS:
+        robot_width = 1 / 2
```
@BergerKing333 Pull these from robot description or just hardcode ig

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
bug fixes

BergerKing333 committed:
bug fixes

umnrobotics committed:
Fixing bugs in autuo dig

umnrobotics committed:
Commenting out so it doesnt move to dig right away

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Merge branch 'main' into autonomy-logic

Isopod00 committed:
Deleted the testing_auto.py file

jmblixt3 committed:
Add rviz config to client since bandwidth cap was removed

jmblixt3 committed:
Ifx namespace conflict
It was possible that the launch configurations could conflict this prevents that form happening

Isopod00 committed:
Update laptop_launch.py

Isopod00 committed:
Update EVERYTHING_launch.py

BergerKing333 committed:
fixing apriltag TF timestamp issue

umnrobotics committed:
comp changes

Isopod00 committed:
Merge branch 'client-rviz' into autonomy-logic

BergerKing333 committed:
fixes

BergerKing333 committed:
final fixes

BergerKing333 committed:
fix

BergerKing333 committed:
fix

umnrobotics committed:
fixed the video device

Isopod00 committed:
Merge branch 'main' into autonomy-logic
